### PR TITLE
feat(theme): add getter and setter for theme user_data

### PR DESCRIPTION
**Pull request by tanguy-rdt**
The `lv_theme_t` struct already exposes a `user_data` field but no
accessor functions were provided, making it impossible to use it
consistently through the public API.

Add `lv_theme_set_user_data()` and `lv_theme_get_user_data()` to allow
callers to attach arbitrary context to a theme instance, following the
same pattern used by other LVGL objects.

### CODE REVIEW

**Review by FASTSHIFT — DISMISSED** [review]

**Review by FASTSHIFT — DISMISSED** [review]

File: src/themes/lv_theme.c [hunk 1]
```diff
@@ -84,6 +84,11 @@ void lv_theme_set_apply_cb(lv_theme_t * theme, lv_theme_apply_cb_t apply_cb)
     theme->apply_cb = apply_cb;
 }
 
+void lv_theme_set_user_data(lv_theme_t * theme, void * user_data)
+{
+    theme->user_data = user_data;
+}
```
Comment (AndreCostaaa) [inline src/themes/lv_theme.c:91]: Check if theme is NULL here before dereferencing

File: src/themes/lv_theme.c [hunk 2]
```diff
@@ -114,6 +119,11 @@ lv_color_t lv_theme_get_color_secondary(lv_obj_t * obj)
     return th ? th->color_secondary : lv_palette_main(LV_PALETTE_BLUE);
 }
 
+void * lv_theme_get_user_data(const lv_theme_t * theme)
+{
+    return theme->user_data;
+}
```
Comment (AndreCostaaa) [inline src/themes/lv_theme.c:127]: Same here: Check if theme is NULL here before dereferencing

File: tests/src/test_cases/test_theme.c [hunk 3]
```diff
@@ -209,4 +209,14 @@ void test_theme_simple(void)
     TEST_ASSERT_NULL(lv_theme_simple_get());
 }
 
+void test_theme_user_data(void)
```
Comment (AndreCostaaa) [inline tests/src/test_cases/test_theme.c:212]: Also add a test checking that NULL doesn't crash the program, you can also add a test that NULL user_data is accepted

**Review by AndreCostaaa — CHANGES_REQUESTED** [review]

File: tests/src/test_cases/test_theme.c (hunk 3 above)
Comment (tanguy-rdt) [reply tests/src/test_cases/test_theme.c:212]: @AndreCostaaa, The changes have been made. Thank you for your feedback.

**Review by FASTSHIFT — DISMISSED** [review]

**Review by kisvegabor — APPROVED** [review]
Thank you! I just rebased it.
